test/git_checkout_test.go

```diff
@@ -0,0 +1,185 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/knative/build-pipeline/test/builder"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	gitSourceResourceName  = "git-source-resource"
+	gitTestTaskName        = "git-check-task"
+	gitTestPipelineName    = "git-check-pipeline"
+	gitTestPipelineRunName = "git-check-pipeline-run"
+)
+
+// TestGitPipelineRun is an integration test that will verify the source code is either fetched or pulled
+// successfully under different revision inputs (branch, commitid, tag, ref)
+func TestGitPipelineRun(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+	t.Parallel()
+
+	revisions := []string{"c15aced0e5aaee6456fbe6f7a7e95e0b5b3b2b2f", "c15aced", "release-0.1", "v0.1.0", "refs/pull/347/head"}
+
+	for _, revision := range revisions {
+
+		c, namespace := setup(t, logger)
+		knativetest.CleanupOnInterrupt(func() { tearDown(t, logger, c, namespace) }, logger)
+		defer tearDown(t, logger, c, namespace)
+
+		logger.Infof("Creating Git PipelineResource %s", gitSourceResourceName)
+		if _, err := c.PipelineResourceClient.Create(getGitPipelineResource(namespace, revision)); err != nil {
+			t.Fatalf("Failed to create Pipeline Resource `%s`: %s", gitSourceResourceName, err)
+		}
+
+		logger.Infof("Creating Task %s", gitTestTaskName)
+		if _, err := c.TaskClient.Create(getGitCheckTask(namespace, t, logger)); err != nil {
+			t.Fatalf("Failed to create Task `%s`: %s", gitTestTaskName, err)
+		}
+
+		logger.Infof("Creating Pipeline %s", gitTestPipelineName)
+		if _, err := c.PipelineClient.Create(getGitCheckPipeline(namespace)); err != nil {
+			t.Fatalf("Failed to create Pipeline `%s`: %s", gitTestPipelineName, err)
+		}
+
+		logger.Infof("Creating PipelineRun %s", gitTestPipelineRunName)
+		if _, err := c.PipelineRunClient.Create(getGitCheckPipelineRun(namespace)); err != nil {
+			t.Fatalf("Failed to create Pipeline `%s`: %s", gitTestPipelineRunName, err)
+		}
+
+		if err := WaitForPipelineRunState(c, gitTestPipelineRunName, timeout, PipelineRunSucceed(gitTestPipelineRunName), "PipelineRunCompleted"); err != nil {
+			t.Errorf("Error waiting for PipelineRun %s to finish: %s", gitTestPipelineRunName, err)
+			taskruns, err := c.TaskRunClient.List(metav1.ListOptions{})
+			if err != nil {
+				t.Errorf("Error getting TaskRun list for PipelineRun %s %s", gitTestPipelineRunName, err)
+			}
+			for _, tr := range taskruns.Items {
+				if tr.Status.PodName != "" {
+					CollectBuildLogs(c, tr.Status.PodName, namespace, logger)
```

Comment:
I don't think the `collectBuildLogs` is adding any value here, we just want to check the `PipelineRunStatus` or the `taskRunStatus` is successful

Reply:
Removed.